plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcConfig.java

```diff
@@ -114,12 +146,21 @@ public BaseJdbcConfig setCacheMaximumSize(long cacheMaximumSize)
         return this;
     }
 
-    @PostConstruct
-    public void validate()
+    @AssertTrue(message = METADATA_CACHE_TTL + " must be set to a non-zero value when " + METADATA_CACHE_MAXIMUM_SIZE + " is set")
+    public boolean isCacheMaximumSizeConsistent()
+    {
+        return metadataCacheTtl.equals(CACHING_DISABLED) ? cacheMaximumSize == BaseJdbcConfig.DEFAULT_METADATA_CACHE_SIZE : true;
+    }
+
+    @AssertTrue(message = METADATA_SCHEMAS_CACHE_TTL + " must not be set when " + METADATA_CACHE_TTL + " is not set")
+    public boolean isSchemaNamesCacheTtlConsistent()
+    {
+        return metadataCacheTtl.equals(CACHING_DISABLED) ? schemaNamesCacheTtl.isEmpty() : true;
```

Comment:
Can we replace with `AssertFalse` and have a condition like `metadataCacheTtl.equals(CACHING_DISABLED) && schemaNamesCacheTtl.isPresent()` or have `AssertTrue` with `!metadataCacheTtl.equals(CACHING_DISABLED)  || schemaNamesCacheTtl.isEmpty()`